test/e2e/e2e-test/src/test/java/org/apache/skywalking/e2e/browser/BrowserE2E.java

```diff
@@ -0,0 +1,353 @@
+                                                                         .setStack("e2e")
+                                                                         .setErrorUrl("/e2e/browser");
+                if (category == ErrorCategory.js) {
+                    errorLogBuilder.setFirstReportedError(true);
+                }
+                sendBrowserErrorLog(errorLogBuilder.build());
+            }
+            return true;
+        } catch (Throwable e) {
+            LOGGER.warn(e.getMessage(), e);
+            return false;
+        }
+    }
+
+    private void sendBrowserPerfData(BrowserPerfData browserPerfData) throws InterruptedException {
+        final CountDownLatch latch = new CountDownLatch(1);
+
+        browserPerfServiceStub.collectPerfData(browserPerfData, new StreamObserver<Commands>() {
+            @Override
+            public void onNext(Commands commands) {
+
+            }
+
+            @Override
+            public void onError(Throwable throwable) {
+                throwable.printStackTrace();
+                latch.countDown();
+            }
+
+            @Override
+            public void onCompleted() {
+                latch.countDown();
+            }
+        });
+        latch.await();
+    }
+
+    private void sendBrowserErrorLog(BrowserErrorLog browserErrorLog) throws InterruptedException {
+        final CountDownLatch latch = new CountDownLatch(1);
+
+        StreamObserver<BrowserErrorLog> collectStream = browserPerfServiceStub.collectErrorLogs(
+            new StreamObserver<Commands>() {
+                @Override
+                public void onNext(Commands commands) {
+
+                }
+
+                @Override
+                public void onError(Throwable throwable) {
+                    throwable.printStackTrace();
+                    latch.countDown();
+                }
+
+                @Override
+                public void onCompleted() {
+                    latch.countDown();
+                }
+            });
+        collectStream.onNext(browserErrorLog);
+        collectStream.onCompleted();
+        latch.await();
+    }
+
+    private void register() {
+        int retryIndex = 0;
+        while (browserId == 0 || browserSingleVersionId == 0) {
+            try {
+                if (browserId == 0) {
+                    browserId = registerBrowser();
+                }
+                if (browserId > 0) {
+                    browserSingleVersionId = registerBrowserSingleVersion(browserId);
+                }
+                if (browserId > 0 && browserSingleVersionId > 0) {
+                    break;
+                }
+            } catch (Throwable e) {
+                if (++retryIndex % 5 == 0) {
+                    LOGGER.error(e.getMessage(), e);
+                }
+            }
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException e) {
+                // ignore
+            }
+        }
+    }
```

Comment:
this should be a test case too, verifying that registering works, and order the cases, place this at the 1st, and the others that depend on this after that, as for ordering cases, refer to `AgentRebootE2E`

Reply:
Thanks @kezhenxu94.  I've changed the code based on the comments :)